plugins/woocommerce-admin/client/products/product-validation.ts

```diff
@@ -51,6 +56,20 @@ export const validate = (
 		);
 	}
 
+	if ( values.sku && values.sku.length > 0 && ! isValidSlug( values.sku ) ) {
+		errors.sku = __(
+			'SKU can only contain alphanumeric characters, hyphens, and underscores.',
+			'woocommerce'
+		);
+	}
+
+	if ( values.sku && values.sku.length > 160 ) {
```

Comment:
I missed that we were changing the SKU validation from what currently exists in core in the original issue. I don't think we should change the validation. Otherwise, merchants will not be able to edit existing products created outside of the new experience (whether via the old experience or by the API) in the new experience.

From https://woocommerce.com/document/managing-products/....

> The SKU field can accept up to 255 characters, which can include alphanumeric values and some special characters. The input is checked using the WordPress function santize_text_field, which means HTML, invalid UTF characters, and octets will be removed when saved (an octet is defined by % followed by 2 characters with a value in the range of a-f or 0-9).

/cc @jarekmorawski @pmcpinto

Reply:
Also, shouldn't we be checking if it is a valid `SKU`, not a valid `slug`?

Reply:
>I don't think we should change the validation. Otherwise, merchants will not be able to edit existing products created outside of the new experience

That's a great point.  I think that alone is enough reason to remove the validation and sanitization from this.  Though we could keep the initial "slug" version on new products.

>Also, shouldn't we be checking if it is a valid SKU, not a valid slug?

Responded to this above, but as you noted, there are not SKU validation rules in core.  We are checking for a valid slug here.

Reply:
All sku validation has been removed from this PR to align with core.